src/components/Image/BaseImage.tsx

```diff
@@ -13,14 +14,23 @@ function BaseImage({onLoad, ...props}: WebImageProps & BaseImageProps) {
             // We override `onLoad`, so both web and native have the same signature
             const {width, height} = event.nativeEvent.source;
             onLoad({nativeEvent: {width, height}});
+
+            if (objectPositionTop) {
+                if (width > height) {
+                    setAspectRatio(1);
+                    return;
+                }
+                setAspectRatio(height ? width / height : 'auto');
+            }
```

Comment:
We shouldn't duplicate this code.

Can't we move the feature implementation to `Image`?